src/main/java/com/github/kagkarlsson/scheduler/Scheduler.java

```diff
@@ -32,12 +32,11 @@
 
 import static com.github.kagkarlsson.scheduler.ExecutorUtils.defaultThreadFactoryWithPrefix;
 
-public class Scheduler implements SchedulerClient {
+public class Scheduler extends SchedulerClient.StandardSchedulerClient {
 
```

Comment:
Ah, you are correct, and I agree with your reasoning. For some reason, that extends still makes me itch :). Another option is implementing the interface, but delegating to an internal `StandardSchedulerClient`. I know it is pretty cosmetic though..